.github/workflows/build.yml

```diff
@@ -22,17 +22,25 @@ jobs:
       WITH_ASAN: ${{ matrix.WITH_ASAN }}
       COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
       COVERALLS_SERVICE_NAME: "GitHub Actions"
+      UPLOAD_COVERALLS: "false"
 
     steps:
       - name: Checkout repository contents
         uses: actions/checkout@v3
 
+      - name: Set RUN_COVERALLS Environment Variable
+        if: env.WITH_ASAN != 'true'
+        # if: (github.event.pull_request.merged == true || github.event_name == 'push') && env.WITH_ASAN != 'true'
+        run: echo "Setting UPLOAD_COVERALLS to true"
+        env:
+          UPLOAD_COVERALLS: true
```

Comment:
This isn't propagated to the next action.
You would need to rather add the following as the "run":
```echo "UPLOAD_COVERALLS=true" >>${GITHUB_ENV}```

But I'm doubtful that will work for pull requests as the COVERALLS_REPO_TOKEN env var will be empty, since secrets are not available for pull requests.

Reply:
Yes sorry, that just updates it for the step. Will set to run on pushes and merged pull requests for now and look at moving to the GitHub Action in the future.